packages/integrations/markdoc/components/RenderNode.astro

```diff
@@ -0,0 +1,30 @@
+---
+import stringifyAttributes from 'stringify-attributes';
+import type { AstroNode } from './astroNode';
+
+type Props = {
+	node: AstroNode;
+};
+
+const Node = (Astro.props as Props).node;
+---
+
+{
+	typeof Node === 'string' ? (
+		<Fragment set:text={Node} />
+	) : 'component' in Node ? (
+		<Node.component {...Node.props}>
+			{Node.children.map((child) => (
+				<Astro.self node={child} />
+			))}
+		</Node.component>
+	) : (
+		<Fragment>
+			<Fragment set:html={`<${Node.tag} ${stringifyAttributes(Node.attributes)}>`} />
+			{Node.children.map((child) => (
+				<Astro.self node={child} />
+			))}
+			<Fragment set:html={`</${Node.tag}>`} />
```

Comment:
Attempted to clean this up attribute stringifier by calling our JSX runtime instead. i.e. `const Component = jsx(Node.tag, Node.attributes)`. However, calling the JSX runtime creates an infinite render loop in development, whether called within the `astroNode.ts` file or the `.astro` frontmatter 😕 Note sure if @natemoo-re has better context on this.

Reply:
Maybe it's good to avoid reusing the JSX runtime for now as according to some recent benchmarks we found that JSX rendering is a bit slow.  https://github.com/withastro/astro/pull/6415

Reply:
Another option would be creating this component with our internal JS API, but this seems fine too.